guide/secure_restful_service/secure_order_mgt_service.bal

```diff
@@ -26,43 +24,53 @@ import ballerinax/kubernetes;
 //    name:"ballerina-guides-secure-restful-service"
 //}
 
+@final string regexInt = "\\d+";
+@final string regexJson = "[a-zA-Z0-9.,{}:\" ]*";
+
 http:AuthProvider basicAuthProvider = {
-    scheme:"basic",
-    authProvider:"config"
+    scheme: "basic",
+    authStoreProvider: "config"
 };
 endpoint http:SecureListener listener {
-    port:9090,
-    authProviders:[basicAuthProvider]
+    port: 9090,
+    authProviders: [basicAuthProvider]
 };
 
 // Order management is done using an in memory map.
 // Add some sample orders to 'orderMap' at startup.
 map<json> ordersMap;
 
-@Description {value:"RESTful service."}
+@Description { value: "RESTful service." }
 @http:ServiceConfig {
-    basePath:"/ordermgt"
+    basePath: "/ordermgt",
+    authConfig: {
+        authentication: { enabled: true }
+    }
 }
 service<http:Service> order_mgt bind listener {
 
-    @Description {value:"Resource that handles the HTTP POST requests that are directed
-     to the path '/orders' to create a new Order."}
+    @Description { value: "Resource that handles the HTTP POST requests that are directed
+     to the path '/orders' to create a new Order." }
     @http:ResourceConfig {
-        methods:["POST"],
-        path:"/order",
-        authConfig:{
-            scopes:["add_order"]
+        methods: ["POST"],
+        path: "/order",
+        authConfig: {
+            scopes: ["add_order"]
         }
     }
     addOrder(endpoint client, http:Request req) {
         json orderReq = check req.getJsonPayload();
         string orderId = orderReq.Order.ID.toString();
+
+        // Get untainted value if `orderId` is a valid input
+        orderId = getUntaintedStringIfValid(orderId);
+
         ordersMap[orderId] = orderReq;
 
         // Create response message.
-        json payload = {status:"Order Created.", orderId:orderId};
+        json payload = { status: "Order Created.", orderId: orderId };
         http:Response response;
-        response.setJsonPayload(untaint payload);
+        response.setJsonPayload(payload);
```

Comment:
Let's use setPayload() here and any other applicable places

Reply:
Fixed.